### ISSUE TITLE
UploadField only shows view button when parent record is editable?

### ISSUE THREAD

**Issue by mlewis-everley** [silverstripe/silverstripe-asset-admin#785]
I have an `Event` object, that can have `many_many` `Attachments` (each of which is a `file`)

I have a user with permission to view `Event` (but not edit)

The user can see the `UploadField` but not the view button (the little eyeball icon)

I have traced this to here: https://github.com/silverstripe/silverstripe-asset-admin/blob/1.0/client/src/components/UploadField/UploadFieldItem.js#L266

Is this right? A user cannot see the "View Button" if they cannot edit?

**Comment by maxime-rainville** [comment]
I've confirmed the issue. This is what the read-only UploadField looks like.

![image](https://user-images.githubusercontent.com/1168676/42144058-15c67c0a-7e0d-11e8-9682-93e7c008bd29.png)

It's not just a question of displaying the view button, we need something to happen when you click said-view-button. This would involved showing some readonly version of `InsertEmbedModal`.

**Comment by mlewis-everley** [comment]
Well if the image/file is readonly, shouldn't it produce the same modal, only readonly?

I know at the moment, if you can edit and then the modal is also read only, but I am assuming that this will change in the future, so if a file/image is readonly, should they just see the current modal?

**Comment by maxime-rainville** [comment]
That modal has a strong assumption that you can pick a different files. It allows you to browse the directories, preview different images, upload new files, create folders, etc. Creating a readonly version that makes sense from a UX perspective might not be a trivial task.

We might also have to consider weird scenarios where a user can edit/upload files, but is not allowed to choose a different record on the current.

**Comment by clarkepaul** [comment]
Agree we need a readonly view, we need to get some designs going to nut out the edge cases, maybe its just a smaller modal with the essentials rather than the whole Files area modal.

**Comment by newleeland** [comment]
**Potiential designs**

**Comment by robbieaverill** [comment]
@newleeland it's worth mentioning that in the backend of silverstripe/framework at the moment there's no differentiation between read only and history view - the history view performs a "readonly transformation" on all the form fields and renders them in that state

**Comment by newleeland** [comment]
> @newleeland it's worth mentioning that in the backend of silverstripe/framework at the moment there's no differentiation between read only and history view - the history view performs a "readonly transformation" on all the form fields and renders them in that state.

Yup ok! The design is mostly to remove the selection of other items readonly view when the uploadfield is opened. So if we don't have another use case for this besides "history" we could potentially just rename this as a future update for history.
